From a3089d00c1f5d57cd503dfcff808e0cc9b65e208 Mon Sep 17 00:00:00 2001
From: Binh Nguyen <binh.nguyen.jz@renesas.com>
Date: Wed, 10 Mar 2021 21:14:10 +0700
Subject: [PATCH 076/135] arm64: dts: rzg2-smarc: Add GPT3 as backlight port.

This commit add GPT channel 3 setting as backlight port.
However, it use same pin with SD1 WP pin. So disable it as default.
CONFIG_BACKLIGHT_PWM is required when want to enable this backlight
setting.

Signed-off-by: LongLuu <long.luu.ur@renesas.com>
---
 arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
index ed0e554..0e77efc 100644
--- a/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
+++ b/arch/arm64/boot/dts/renesas/rzg2-smarc.dtsi
@@ -77,6 +77,14 @@
 			read-only;
 		};
 	};
+
+	backlight {
+		compatible = "pwm-backlight";
+		pwms = <&gpt3 0 50000>;
+
+		brightness-levels = <0 2 8 16 32 64 128 255>;
+		default-brightness-level = <6>;
+	};
 };
 
 &xinclk {
@@ -198,6 +206,12 @@
 		groups = "rspi1_clk_b", "rspi1_data_b", "rspi1_ssl_b";
 		function = "rspi1";
 	};
+
+	gpt3_pins: gpt3 {
+		groups = "gpt3_b_b";
+		function = "gpt3";
+	};
+
 };
 
 &scif0 {
@@ -355,3 +369,10 @@
 	pinctrl-names = "default";
 	status = "okay";
 };
+
+&gpt3 {
+	pinctrl-0 = <&gpt3_pins>;
+	pinctrl-names = "default";
+	//conflict pin with SD1 WP, disable as default
+	//status = "okay";
+};
-- 
2.7.4

